ユーザ管理機能実装

matyareo-matyarei opened the pull request:
# What
ユーザー管理機能の実装
新規登録のGyazo GIF：　https://gyazo.com/491c75d0c727be3789cb5cb64fbea35c
テストコードGyazo：　https://gyazo.com/ed4aca9577b8872557d9da5233c1a793
ログイン・ログアウトの挙動のGIF　https://gyazo.com/b879f66cb36129e81e78bd830bcc1cdf
エラーメッセージの表示画像　https://gyazo.com/e05d5160103e19008a119020eae6905a

# Why
ユーザー管理をするため

matyareo-matyarei committed:
first commit

トップページのみ設定済

matyareo-matyarei committed:
deviseの導入、usersテーブル作成

matyareo-matyarei committed:
ユーザー登録機能実装

matyareo-matyarei committed:
ユーザーログイン機能実装

matyareo-matyarei committed:
ユーザー登録機能の修正

matyareo-matyarei committed:
Rspec導入

matyareo-matyarei committed:
エラーハンドリング設定

matyareo-matyarei committed:
Userモデルの単体テストコード

モデルのバリデーションも書き直しをした。

ky3090 commented on app/models/user.rb:
```diff
@@ -0,0 +1,20 @@
+  validates_format_of :password, with: PASSWORD_REGEX, message: 'Include both letters and numbers'
+
+  validates :nickname, presence: true
+  validates :birthday, presence: true
```
9,10行目のvalidationもwith_optionsでまとめましょう！

ky3090 commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,109 @@
+      expect(@user.errors.full_messages).to include("Password is too short (minimum is 6 characters)")
+    end
+
+    it "passwordが半角英数字を混合していないと登録できない" do
```
数字のみの場合登録できないことのテストも行いましょう！

ky3090 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

ky3090 commented:
キャプチャ画像ありがとうございます！
ログインの挙動の画像とエラーメッセージが表示される際のキャプチャ画像をいただけますと幸いです。
よろしくお願いいたします。

matyareo-matyarei commented:
失礼しました。
修正致しましたので、ご確認お願いします。

matyareo-matyarei committed:
ユーザー管理機能

すいません、１回目のプルリクエスト時にRubocopの実行を忘れておりました
変更点
・バリデーションをwith_optionでまとめた
・passwordのテストコードに数字のみの場合も追記した

chirikuzu commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,112 @@
+      expect(@user.errors.full_messages).to include('Password is too short (minimum is 6 characters)')
+    end
+
+    it 'passwordが半角英数字を混合していないと登録できない' do
+      @user.password = 'aaaaaa'
+      @user.password_confirmation = 'aaaaaa'
+      @user.valid?
+      expect(@user.errors.full_messages).to include('Password Include both letters and numbers')
+      @user.password = '000000'
+      @user.password_confirmation = '000000'
+      @user.valid?
+      expect(@user.errors.full_messages).to include('Password Include both letters and numbers')
+    end
```
こちら英字のみの場合と数字のみの場合でit句を分けましょう。
一つずつ検証を行うことでテストの正確性が向上します。

chirikuzu commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,112 @@
+      expect(@user.errors.full_messages).to include("First name can't be blank")
+    end
+
+    it 'ユーザー本名は、全角（漢字・ひらがな・カタカナ）での入力でないと登録できない' do
+      @user.family_name = 'aaa'
+      @user.first_name = 'aaa'
+      @user.valid?
+      expect(@user.errors.full_messages).to include('Family name Full-width characters', 'First name Full-width characters')
```
こちらはfamily_name、first_nameでそれぞれit句を分けましょう。
1つのit句内では1つのカラムに対してテストを行うことでテストの正確性が向上します。

chirikuzu commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,112 @@
+      expect(@user.errors.full_messages).to include("First kana can't be blank")
+    end
+
+    it 'ユーザー本名のフリガナは、全角（カタカナ）での入力でないと登録できない' do
+      @user.family_kana = 'ひらがな'
+      @user.first_kana = 'ひらがな'
+      @user.valid?
+      expect(@user.errors.full_messages).to include('Family kana Full-width katakana characters', 'First kana Full-width katakana characters')
+    end
```
こちらについても同様にfamily_kanaおよびfirst_kanaでit句を分けましょう。

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

matyareo-matyarei committed:
ユーザー管理機能修正(3)

ky3090 reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！
